### DISCUSSION TITLE
Trouble with protocols

### DISCUSSION THREAD

**Discussion by NickNoble-Slalom** [hmlongco/Factory#188] [Q&A]
I have a protocol called `Routable` and a class implementation called `Router`. I would like to register this in my `Container`, then overwrite it for unit testing. This is what I have so far:

```swift
extension Container {
    var myRouter: Factory<some Routable> { self { Router() }.singleton }
}
```

And in my unit tests:
```swift
class MockRouter: Routable {
...
}
```

```swift
Container.shared.myRouter.register { MockRouter() }
```
with this error:
`Cannot convert value of type 'MockRouter' to closure result type 'some Routable'`

What am I doing wrong here?

**Comment by hmlongco** [comment]
If it's just a protocol then just do `Factory<Routable>`. If Routable has some sort of associated type (some?) then you can't do that as there's no way for Swift to determine the concrete type behind the closure.

**Reply by NickNoble-Slalom** [reply]
What do you mean by "just do Factory"?

**Reply by hmlongco** [reply]
HTML ate the braces, sorry.
